python/lsst/cell_coadds/_fits.py

```diff
@@ -374,11 +432,59 @@ def writeMultipleCellCoaddAsFits(
     metadata : `~lsst.daf.base.PropertySet`, optional
         Additional metadata to write to the FITS file.
 
+    Returns
+    -------
+    hdu_list : `~astropy.io.fits.HDUList`
+        The FITS file as an HDUList.
+
     Notes
     -----
     Changes to this function that modify the way the file is written to disk
     must be accompanied with a change to FILE_FORMAT_VERSION.
     """
+    # Create metadata tables:
+    # 1. Visit table containing information about the visits.
+    # 2. Cell table containing info about the visit+detector for each cell.
+    visit_records: list[Any] = []
+    cell_records: list[Any] = []
+    instrument_set = set()
+    for cell_id, single_cell_coadd in multiple_cell_coadd.cells.items():
+        for observation_id in single_cell_coadd.inputs:
+            visit_records.append(
+                (observation_id.visit, observation_id.physical_filter, observation_id.day_obs)
+            )
+            cell_records.append(
+                (cell_id.x, cell_id.y, observation_id.visit, observation_id.detector, observation_id.packed)
+            )
+            instrument_set.add(observation_id.instrument)
+
+    assert len(instrument_set) == 1, "All cells must have the same instrument."
+    instrument = instrument_set.pop()
+
+    visit_recarray = np.rec.fromrecords(
```

Comment:
Several years ago, I remember learning that numpy record arrays had some surprising performance overheads compared to using a regular `numpy.ndarray` with a structured dtype, and I've avoided them ever since, since the difference is just whether you use attribute access vs. item access (i.e. `row["field_name"]`).  I have no idea if that's still the case.

Reply:
I don't care as much about the attribute access vs. item access, but avoiding positional index access and also having to create several `fits.Column` objects. I'm inclined to leave this as is, and change it in the future if there is a noticeable overhead. Fortunately, this is purely internal and can be changed without affecting `FILE_FORMAT_VERSION` at all.

Reply:
:+1:, it sounds like astropy does something special with `numpy.recarray`, which is not something I realized.